### ISSUE TITLE
[iOS] Framework Search paths are wiped clean on RN link

### ISSUE THREAD

**Issue by ssomnoremac** [facebook/react-native#10546]
### Description & Reproduction

run `react-native link react-native-fbsdk` and the framework search paths that I added for react-native-lock are wiped out. Using frameworks to make lock work with fcm by the way.

* React Native version: 35
* Platform: iOS
* Operating System: MacOS

**Comment by lacker** [comment]
I'm not sure what you mean by "the framework search paths that you added for react-native-lock". Are you referring to the part of the instructions in https://github.com/auth0/react-native-lock where it says "This more of a hack than an actual solution."?

Basically it is not clear to me that this is supposed to work. Does it work if you just install `react-native-lock` and `react-native-fbsdk` in the normalest way possible?

**Comment by ssomnoremac** [comment]
I'm saying that I can't use link to install packages anymore because, yes,
I guess it wipes out this 'hack'. I guess since it's a hack so I shouldn't
expect it to persist in link. Maybe just close this then.

On Oct 26, 2016 12:58 AM, "Kevin Lacker" <notifications@github.com> wrote:

I'm not sure what you mean by "the framework search paths that you added
for react-native-lock". Are you referring to the part of the instructions
in https://github.com/auth0/react-native-lock where it says "This more of a
hack than an actual solution."?

Basically it is not clear to me that this is supposed to work. Does it work
if you just install react-native-lock and react-native-fbsdk in the
normalest way possible?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/facebook/react-native/issues/10546#issuecomment-256249616>,
or mute the thread

**Comment by lacker** [comment]
OK. Do you know why you had to install react-native-lock in this way in the first place? Perhaps the right answer is to improve react native's ability to install things without mucking around with framework search paths.

**Comment by ssomnoremac** [comment]
Sure, I needed to do this because I was using react-native-fcm which requires the `uses_framework` tag to be uncommented. So for a reason unknown to me, this breaks the standard library installation of react-native-lock and requires the 'hack'. I'm not an iOS dev so I'm not sure what the deal is.

**Comment by lacker** [comment]
This library? - https://github.com/evollu/react-native-fcm I was trying to find in their docs why it requires `uses_framework!` but those docs don't seem to refer to it.

**Comment by ssomnoremac** [comment]
You're right. Honestly, I've make so many adjustments to get fcm to work with lock that I can't tell you when I first uncommented `uses_framework!` Maybe I need to go back and try and reinstall both libraries without frameworks to see again. Not sure when I'll have time for that though.
